extracted/plugins/SocketPlugin/src/common/SocketPlugin.c

```diff
@@ -2306,11 +2306,11 @@ socketRecordSize(void)
 static SocketPtr
 socketValueOf(sqInt socketOop)
 {
-	return ((isBytes(socketOop))
-	 && ((byteSizeOf(socketOop)) == (sizeof(SQSocket)))
-		? ((SocketPtr) (firstIndexableField(socketOop)))
-		: (primitiveFailFor(PrimErrBadArgument),
-			null));
+	if ((isBytes(socketOop)) && ((byteSizeOf(socketOop)) == (sizeof(SQSocket)))) {
+		return (SocketPtr) (firstIndexableField(socketOop));
+	}
+	primitiveFailFor(PrimErrBadArgument);
+	return null;
```

Comment:
Why would this rewrite remove the warning? do you know?

Reply:
Because there is not the expression with the complex type I guess (split into many statements)